git_repo/services/service.py

```diff
@@ -32,8 +32,8 @@
 
 class ProgressBar(RemoteProgress): # pragma: no cover
     '''Nice looking progress bar for long running commands'''
-    def setup(self, repo_name):
-        self.bar = Bar(message='Pulling from {}'.format(repo_name), suffix='')
+    def setup(self, repo_name, msg='Pulling from {}'):
```

Comment:
instead of having custom messages being added by services, the constructor could take an 'action' argument which was an enum like `PULL: 1; PUSH: 2` with PULL being the default to maintain backwards compatibility (so the rest of the code doesnt need to change).  Then the two messages would be kept inside this class.

Reply:
Okay. I will do it that way